.github/workflows/deploy.yml

```diff
@@ -0,0 +1,69 @@
+---
+name: CI docs
+
+on:
+    push:
+        branches:
+            - master
+
+jobs:
+    build-pages:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v3
+              with:
+                  submodules: true
+
+            - name: Set up Python
+              uses: actions/setup-python@v4
+              with:
+                  python-version: '3.10'
+                  cache: pip
+                  cache-dependency-path: |
+                      requirements*.txt
+
+            - name: Install dependencies
+              run: |
+                  python -m pip install --upgrade pip
+                  pip install -U sphinx~=7.2 sphinx-rtd-theme~=1.3 myst-parser~=2.0
+                  pip install -r requirements.txt
+                  pip freeze
+
+            - name: Build FileType docs
+              run: |
+                  gen-doc schemas/filetype.yml -d docs/source/gen-doc
+
+            - name: Build Extractor docs
+              run: |
+                  gen-doc schemas/extractor.yml -d docs/source/gen-doc
+
+            - name: Build Sphinx docs
+              run: |
+                  sphinx-build -b html docs/source public/master
+
+            - name: Upload docs artifact
+              uses: actions/upload-artifact@v2
+              with:
+                  name: public-master
+                  path: public/master
+    deploy-pages:
+        runs-on: ubuntu-latest
+        needs:
+            - build-pages
+        steps:
+            - uses: actions/checkout@v3
+
+            - uses: actions/setup-python@v4
+              with:
+                  python-version: '3.9'
+
+            - uses: actions/download-artifact@v2
+              with:
+                  name: public-master
+                  path: public/master
+
+            - uses: peaceiris/actions-gh-pages@v3
+              with:
+                  github_token: ${{ secrets.GITHUB_TOKEN }}
+                  publish_dir: ./public
+                  keep_files: true
```

Comment:
There is an official action for this (in beta): https://github.com/actions/upload-pages-artifact

Here's what I use on another project:

```
    - name: Upload docs artifact
      if: matrix.python-version == 3.9
      uses: actions/upload-pages-artifact@v2
      with:
        path: site
```

Looks like it might also need another step to deal with file permissions though: https://github.com/actions/upload-pages-artifact?tab=readme-ov-file#file-permissions

If this one is working and is still maintained then lets keep it!

Reply:
I'll leave this one in until it fails...